tools/ci_build/github/azure-pipelines/templates/react-native-ci.yml

```diff
@@ -99,17 +98,15 @@ stages:
   jobs:
   - job: ReactNative_CI
     pool:
-      # We need macOS-12 to run the Android emulator for now.
-      # https://github.com/actions/runner-images/issues/7671
-      vmImage: 'macOS-12'
+      vmImage: 'macOS-13'
     variables:
       runCodesignValidationInjection: false
     timeoutInMinutes: 90
     steps:
     - task: mspremier.PostBuildCleanup.PostBuildCleanup-task.PostBuildCleanup@3
       displayName: Clean Agent Directories
       condition: always()
-
+    - template: use-xcode-version.yml
```

Comment:
The default version of xcode from MacOS-13 is 15.2 which has features that was deprecated from 
```
/boost/container_hash/hash.hpp:131:33: error: no template named 'unary_function' in namespace 'std'; did you mean '__unary_function'?
        struct hash_base : std::unary_function<T, std::size_t> {};
                           ~~~~~^~~~~~~~~~~~~~
                                __unary_function

```
So, I am using the Xcode 14.3.1 like every other places does

Reply:
May be able to update the boost version to address that. See comment on initial post here: https://stackoverflow.com/questions/77133361/no-template-named-unary-function-in-namespace-std-did-you-mean-unary-fun

> React Native and boost have now fixed there source. Boost use BOOST_NO_CXX98_FUNCTION_BASE to block using these deprecated functions. Unfortunately by default boost tries to guess if your compile has them rather than turning them off in c++17 and above as it should.